src/coreclr/jit/liveness.cpp

```diff
@@ -932,8 +947,20 @@ bool Compiler::fgComputeLifeTrackedLocalDef(VARSET_TP&           life,
             }
 #endif // DEBUG
         }
+        else
+        {
+            // If the only uses are useasg defs, then the variable is not live unless artificially kept alive.
+            isLive =
+                VarSetOps::IsMember(this, fgTrueUseSet, varIndex) || VarSetOps::IsMember(this, keepAliveVars, varIndex);
```

Comment:
If I understand right the `fgTrueUseSet` computed is a global set of variables and is not computed via data flow. Do you know how conservative that is, compared to making the data flow computation more precise?

I would be interesting in which liveness phase (early liveness, SSA's liveness or lowering's liveness) actually end up removing these stores. If it is not removed by SSA's liveness then I think we can do better by making the liveness computations more precise (similar to physical promotion's liveness).

Reply:
Just out of curiosity I opened a PR for my old change: https://github.com/dotnet/runtime/pull/115081

Reply:
Spot checking I see some during SSA and some during Lower.

Let me add some metrics and see if I can get a more comprehensive picture.

Reply:
#115081 does seem to get most of the same diffs and a few more. Given that it does not require any new accounting I would be inclined to suggest that approach instead. I would also imagine it does not have any TP diffs once things are templated correctly, so it does require a bit clean up.

Reply:
Fine by me. Are you going to finish off your version?

Reply:
Yeah, let me try to do that.